### PR TITLE
Productionize doctor visits

### LINKED ISSUE #1676: Productionize-doctor-visits
Productionize doctor visitis

- Switch from print() to logger
- Configure delphi_utils.validator to replicate as much of Maria's validation routines as is feasible

What to do about data fetching:
* The current doctor-visits pipeline wraps everything in a shell script.
* The current covidcast-indicators delphi_utils.runner pipeline uses Python for this.
* Move everything into delphi_utils.runner: -requires more development effort to add data fetching, +makes pipelines uniform
   * [x] refactor data fetching
   * [x] refactor secrets into params.json
   * [x] Automation step calls `python -m delphi_utils.runner delphi_doctor_visits`

#1677

### PR DESCRIPTION
**Pull request by jingjtang**
### Description
Update the current pipeline to deploy `doctor_visits` like an indicator while keeping intact its current automation routines.

### Changelog
- Adds additonal directory structure and current automation scripts.
- Gives Ansible the ability to place a templated `secrets.py` file into an indicator's root directory.
- Updates vars and vault with new secrets.
- All the pipeline required code are moved to `delphi_doctor_visits`. 
- Switch from print() to logger
- Add pulling steps from shell scripts to `run.py`
- Pulled files are downloaded and stored in `./retrieve_files`. All the raw files will be deleted after a production run
- refactor secrets into params.json
- Check the validation results. The type of hrr geo_ids are preferred to be string. However, if I change it to zero-filled strings, it does not match the history which generates more problems. So, switched back. 

TODO
* [x] check  whether `max_dates` and  `n_csv_export` work as expected
* [x] check with Maria what to do with 2 drops a day
### Fixes
Fixes #1676

### CODE REVIEW

**Comment by jingjtang** [comment]
Get replay from Maria:

_Hi Jingjing, awesome to hear about the production work!

The two drops are the same type of data (i.e. they are generated through identical queries/scripts), just run at different times of the day. Sometimes they miss one, so we only get one. Sometimes they miss both :/

But generally they both come, and there's a cron job to run the indicator generation in the evening after the second drop. The script should just look for whichever drop is latest (so the most recent one), and ignore the earlier one, if it exists. 

Best,
Maria_

File: ansible/vars.yaml [hunk 1]
```diff
@@ -32,6 +32,12 @@ claims_hosp_ftp_password: "{{ vault_claims_hosp_ftp_password }}"
 claims_hosp_ftp_host: "{{ vault_claims_hosp_ftp_host }}"
 claims_hosp_midas_user: "{{ vault_claims_hosp_midas_user }}"
 claims_hosp_midas_password: "{{ vault_claims_hosp_midas_password }}"
+# doctor_visits
+claims_hosp_ftp_user: "{{ vault_doctor_visits_ftp_user }}"
```
Comment (korlaxxalrok) [inline ansible/vars.yaml:36]:
```suggestion
doctor_visits_ftp_user: "{{ vault_doctor_visits_ftp_user }}"
```

File: ansible/vars.yaml [hunk 2]
```diff
@@ -32,6 +32,12 @@ claims_hosp_ftp_password: "{{ vault_claims_hosp_ftp_password }}"
...
+claims_hosp_ftp_password: "{{ vault_doctor_visits_ftp_password }}"
```
Comment (korlaxxalrok) [inline ansible/vars.yaml:37]:
```suggestion
doctor_visits_ftp_password: "{{ vault_doctor_visits_ftp_password }}"
```

File: ansible/vars.yaml [hunk 3]
```diff
@@ -32,6 +32,12 @@ claims_hosp_ftp_password: "{{ vault_claims_hosp_ftp_password }}"
...
+claims_hosp_ftp_host: "{{ vault_doctor_visits_ftp_host }}"
```
Comment (korlaxxalrok) [inline ansible/vars.yaml:38]:
```suggestion
doctor_visits_ftp_host: "{{ vault_doctor_visits_ftp_host }}"
```

File: ansible/vars.yaml [hunk 4]
```diff
@@ -32,6 +32,12 @@ claims_hosp_ftp_password: "{{ vault_claims_hosp_ftp_password }}"
...
+claims_hosp_midas_user: "{{ vault_doctor_visits_midas_user }}"
```
Comment (korlaxxalrok) [inline ansible/vars.yaml:39]:
```suggestion
doctor_visits_midas_user: "{{ vault_doctor_visits_midas_user }}"
```

File: ansible/vars.yaml [hunk 5]
```diff
@@ -32,6 +32,12 @@ claims_hosp_ftp_password: "{{ vault_claims_hosp_ftp_password }}"
...
+claims_hosp_midas_password: "{{ vault_doctor_visits_midas_password }}"
```
Comment (korlaxxalrok) [inline ansible/vars.yaml:40]:
```suggestion
doctor_visits_midas_password: "{{ vault_doctor_visits_midas_password }}"
```

File: doctor_visits/delphi_doctor_visits/download_claims_ftp_files.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    client = paramiko.SSHClient()
+    client.set_missing_host_key_policy(AllowAnythingPolicy())
+
+    client.connect(ftp_credentials["host"],
+                   username=ftp_credentials["user"],
+                   password=ftp_credentials["pass"],
+                   port=ftp_credentials["port"])
+    sftp = client.open_sftp()
+    sftp.chdir('/optum/receiving')
+    #sftp.chdir('/mjahja/optum/receiving_aggregated_20200602_20200819')
```
Comment (korlaxxalrok) [inline doctor_visits/delphi_doctor_visits/download_claims_ftp_files.py:70]: Suggest clean up comment if unused

**Review by korlaxxalrok — APPROVED** [review]
Looks like a good start!

**Pull request merged**